server/static/editor-plugins/addons.js

```diff
@@ -1,10 +1,99 @@
 // https://h5p.org/adding-text-editor-buttons
 
+function getFixMathTypeDialog() {
+  let offsetTop = 0;
+  let viewportHeight = 0;
+  let parentOffsetTop = 0;
+  let parentWidth = 0;
+  let parentHeight = 0;
+  let mathTypeInstances = [];
+  let waitingForUpdate = false;
+
+  const setMathTypePosition = () => {
+    const visibleInstances = mathTypeInstances.filter(
+      (editor) => !editor.classList.contains('wrs_closed')
+    );
+    if (waitingForUpdate || visibleInstances.length === 0) return;
+    waitingForUpdate = true;
+    requestAnimationFrame(() => {
+      try {
+        const boundingRects = visibleInstances
+          .map((dialog) => dialog.getBoundingClientRect())
+        const instanceHeights = boundingRects.map((rect) => rect.height)
+        const totalHeight = instanceHeights
```

Comment:
Why sum up all the heights of all the mathType instances? I thought there was at most one dialog open per page

Reply:
I thought so too until I realized two things:
1. There **_can_** be more than one instance of MathType open at a time. Prior to this change that was unclear because the dialogs would stack up in one place.
2. There is a source editor that should be moved too.

The video demos both of these cases if you want to see.